ui/src/app/views/conceptset-create-modal/component.tsx

```diff
@@ -0,0 +1,122 @@
+import {Component, Input} from '@angular/core';
+import {AlertDanger} from 'app/components/alert';
+import {Button} from 'app/components/buttons';
+import {TextInput, ValidationError} from 'app/components/inputs';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {WorkspaceData} from 'app/resolvers/workspace';
+import {conceptSetsApi} from 'app/services/swagger-fetch-clients';
+import {ReactWrapperBase, summarizeErrors, withCurrentWorkspace} from 'app/utils/index';
+import {ConceptSet, CreateConceptSetRequest, Domain, DomainInfo} from 'generated/fetch';
+import * as React from 'react';
+import {validate} from 'validate.js';
+
+
+export const CreateConceptSetModal = withCurrentWorkspace()
+(class extends React.Component<{
+  workspace: WorkspaceData,
+  onCreate: Function,
+  onClose: Function,
+  conceptDomainList: Array<DomainInfo>,
+  existingConceptSets: ConceptSet[],
+}, {
+  name: string,
+  description: string,
+  domain: Domain,
+  nameTouched: boolean,
+  saving: boolean,
+  savingError: string,
+}> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      description: '',
+      domain: props.conceptDomainList[0] as unknown as Domain,
+      nameTouched: false,
+      saving: false,
+      savingError: ''
+    };
+  }
+
+  async saveConceptSet() {
+    try {
+      const {name, description, domain} = this.state;
+      const {workspace} = this.props;
+      const concept: ConceptSet = {
+        name: name,
+        description: description,
+        domain: domain
+      };
+      const request: CreateConceptSetRequest = {
+        conceptSet: concept
+      };
+      this.setState({saving: true});
+      await conceptSetsApi().createConceptSet(workspace.namespace, workspace.id, request);
+      this.props.onCreate();
+      this.props.onClose();
+    } catch (e) {
+      console.error(e);
+      this.setState({savingError: e});
```

Comment:
I think this will end up saving the response object itself, which probably isn't what we want. Since this is being used as a boolean, perhaps it should just be one.